lightning/src/ln/channel.rs

```diff
@@ -5119,8 +5119,46 @@ impl<Signer: WriteableEcdsaChannelSigner> Channel<Signer> {
 						return Ok((Some(channel_ready), announcement_sigs));
 					}
 				}
+
+				// If we have a vanilla LN channel, this checks if the transaction
+				// spends from the actual funding output.
+				//
+				// If we have a split channel, this checks if the transaction spends
+				// from the glue output.
+				//
+				// In both cases we are checking if this transaction is the
+				// commitment transaction.
+				let is_funding_or_glue_txo = |prev_outpoint: &bitcoin::OutPoint| -> bool {
+					prev_outpoint == &funding_txo.into_bitcoin_outpoint()
+				};
+
+				// This check only runs if the check above returns `false`.
+				//
+				// Given that, if we have a vanilla LN channel this check will
+				// return `false` too, because for vanilla LN channels `funding_txo
+				// == original_funding_txo`.
+				//
+				// If we have a split channel, this checks if the transaction spends
+				// from the actual funding output BUT it is _not_ the split
+				// transaction. This can happen if the split channel is closed using
+				// a revoked commitment transaction!
```

Comment:
Does this make sense? I don't see any other way that we could spend the split channel's funding output if it's not a split transaction.

Reply:
It could also potentially be a mutual close, though that's not properly supported at the moment.

Reply:
Ah, that makes sense too. I can include that in the comment.

Reply:
> It could also potentially be a mutual close, though that's not properly supported at the moment.

We've recently come to the conclusion that this would be a really interesting feature for 10101, particularly in those situations where we encounter problems with the subchannel i.e. the usual problem where we are stuck in an intermediate state of either opening or closing the subchannel.

What's hard is to agree on how the money should be split, because we don't have the full context in LDK i.e. what if there is a DLC to be settled. I was thinking that maybe we could propose the values from the outside, but that would be a bigger change in LDK.